channel/channel.go

```diff
@@ -0,0 +1,57 @@
+package channel
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"time"
+)
+
+type CryChan[T any] struct {
```

Comment:
If the package is called `channel`, then the struct should be `Cry`. Like so we get to use a `channel.Cry`.

Reply:
naming wise, wouldn't it make sense to call it `WithTimeout` for clarity (i.e. `channel.WithTimeout`), or something implying its use `Debug` ? 

`Cry` is a bit obscure without further context😅